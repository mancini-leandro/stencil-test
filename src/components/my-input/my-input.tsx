import { Component, Host, h, EventEmitter, Event, Prop } from '@stencil/core';

@Component({
  tag: 'my-input',
  styleUrl: 'my-input.css',
  shadow: true,
})
export class MyInput {

  @Prop({ mutable: true })
  value: string;

  @Event()
  inputValueChange: EventEmitter<string>;

  onInputChanged = (ev) => {
    this.inputValueChange.emit(ev.target.value);
  }

  render() {
    return (
      <Host>
        <input
          type="text"
          onInput={this.onInputChanged}
          value={this.value} />
      </Host>
    );
  }

}
