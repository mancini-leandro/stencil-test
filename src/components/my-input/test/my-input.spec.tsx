import { newSpecPage } from '@stencil/core/testing';
import { MyInput } from '../my-input';

describe('my-input', () => {
  it('should input value', async () => {
    const page = await newSpecPage({
      components: [MyInput],
      html: `<my-input value="Leandro Mancini"></my-input>`
    });

    const input = page.root.shadowRoot.querySelector('input');

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
    expect(input.value).toEqual('Leandro Mancini');
  });

  it('should emit value when input entered', async () => {
    const page = await newSpecPage({
      components: [MyInput],
      html: `<my-input></my-input>`
    });
    const spyInput = jest.fn();
    const input = page.root.shadowRoot.querySelector('input');

    page.doc.addEventListener('inputValueChange', spyInput);

    input.value = 'Leandro';

    await input.dispatchEvent(new Event('input'));
    await page.waitForChanges();

    expect(spyInput).toHaveBeenCalled();
    expect(spyInput.mock.calls[0][0].detail).toEqual('Leandro');
  });
});
