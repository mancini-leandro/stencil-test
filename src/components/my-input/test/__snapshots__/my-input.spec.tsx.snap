// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-input should input value 1`] = `
<my-input value="Leandro Mancini">
  <mock:shadow-root>
    <input type="text" value="Leandro Mancini">
  </mock:shadow-root>
</my-input>
`;
